### ISSUE TITLE
Add license to Javascript files

### ISSUE THREAD

**Issue by kartikynwa** [LemmyNet/lemmy#1063]
This would make Lemmy compatible with LibreJS.

[Originally mentioned in this post.](https://dev.lemmy.ml/post/38598/comment/14472) The methods how to do that are outlined [in the GNU LibreJS documentation](https://www.gnu.org/software/librejs/free-your-javascript.html#magnet-link-license).

@dessalines [comment](https://dev.lemmy.ml/post/38598/comment/14475) on that post:

> That seems pretty easy to add, this site uses AGPLv3 so we could add a magnet link in there. Sounds like a good first PR.

**Comment by kartikynwa** [comment]
I could do this but I don't know which files the license needs to be added to. If you could let me know, I can make some progress on this.

**Comment by dessalines** [comment]
I took a pass at this and its actually kind of difficult without some manual work. We're using `fuse-box` as our bundler (it bundles all your javascript into a single file, and minifies and treeshakes it), which has a [Banner plugin](https://fuse-box.org/docs/3.5.0/plugins/banner-plugin#description) (IE it adds something to the top of your js bundle).

The problem is its using this plugin called Quantum, which seems to overwrite the banner, and I couldn't get it to write it out for the prod builds. Check out the branch I just pushed up called `libre_js_license`.

**Comment by kartikynwa** [comment]
I just asked about this in fuse-box's slack. Let's see if I get a reply.

**Comment by kartikynwa** [comment]
This is what I got @dessalines 

![screenshot-20200820185653](https://user-images.githubusercontent.com/7846060/90813349-199a1b80-e345-11ea-8f9e-df8973788ce8.png)

**Comment by dessalines** [comment]
Yeah, unfortunately we do have an open issue to move away from fuse-box, I could def use some help with it. I'm trying out parcel right now, but it doesn't seem to work with inferno's `BrowserRouter`: https://github.com/jayy-lmao/inferno-parcel-ts/issues/5

**Comment by fifn2** [comment]
I would trust the JS on Lemmy a lot more if it was LibreJS compliant.
